app/javascript/components/users/SignInComponent.tsx

```diff
@@ -36,18 +47,22 @@ function SignInComponent(props:SignInComponentProps) : JSX.Element {
 		if (failed && wasSubmitting) {
 			// we JUST failed so we only call onFailure
 			// once
-			onFailure(lastError);
+			setComponentState('ready');
+			props.onFailure(lastError);
 		}
 
-		if (wasSubmitting && !failed){
+		if (wasSubmitting && !failed) {
 			// we JUST succeeded
-			// TODOtestid
+			// TODO
+			setComponentState('success');
+			props.onSuccess();
 		}
-	}, [failed, submitting, previousSubmittingValue, onFailure, lastError]);
+	}, [failed, submitting, previousSubmittingValue, lastError]);
```

Comment:
`yarn eslint --fix` will tell you there's a bug here.

Reply:
I have fixed this bug!